main.js

```diff
@@ -0,0 +1,190 @@
+let recipes = []; 
+let ingredientCount = 5; 
+
+async function fetchRecipes() {
+  try {
+    const response = await fetch('https://raw.githubusercontent.com/antuanettaV/antuanettaV.github.io/main/test.json'); 
+    if (!response.ok) throw new Error('Network response was not ok');
+    recipes = await response.json();
+    displayRecipes(recipes);
+  } catch (error) {
+    console.error('Error fetching recipes:', error);
+  }
+}
+
+function displayRecipe(recipe) {
+  const container = document.getElementById("recipe-container");
+  const recipeElement = document.createElement("div");
+  recipeElement.className = "recipe";
+
+  const title = document.createElement("h2");
+  title.textContent = recipe.name;
+  recipeElement.appendChild(title);
+
+  const description = document.createElement("p");
+  description.textContent = recipe.description;
+  recipeElement.appendChild(description);
+
+  const img = document.createElement("img");
+  img.src = recipe.pictureUrl;
+  img.alt = recipe.name;
+  recipeElement.appendChild(img);
+
+  const ingredientsTitle = document.createElement("h3");
+  ingredientsTitle.textContent = "Ingredients:";
+  recipeElement.appendChild(ingredientsTitle);
+
+  const ingredientsList = document.createElement("ul");
+  recipe.ingredients.forEach((ingredient) => {
+    const listItem = document.createElement("li");
+    listItem.textContent = ingredient;
+    ingredientsList.appendChild(listItem);
+  });
+  recipeElement.appendChild(ingredientsList);
+
+  container.appendChild(recipeElement);
+}
+
+function displayRecipes(recipeArray) {
+  const recipeGrid = document.getElementById("recipe-grid");
+  recipeGrid.innerHTML = "";
+
+  recipeArray.forEach((recipe) => {
+    const recipeCard = document.createElement("div");
+    recipeCard.className = "recipe-card";
+    recipeCard.innerHTML = `
+      <h2>${recipe.name}</h2>
+      <p>Ingredients:</p>
+      <ul>
+        ${recipe.ingredients.map((ingredient) => `<li>${ingredient}</li>`).join("")}
+      </ul>
+    `;
+    recipeGrid.appendChild(recipeCard);
+  });
+}
+
+async function addNewRecipe(newRecipe) {
+  try {
+    const response = await fetch('https://raw.githubusercontent.com/antuanettaV/antuanettaV.github.io/main/test.json', { 
```

Comment:
Making a POST call to your API is a great idea. This is how it would be done in a professional scenario. However, please note that you are not actually saving the new recipe in your API because the JSON file remains the same. You will learn more about this in the coming weeks at Hack Your Future. But for now, I recommend you omit the POST call because it creates errors in your application:

<img width="574" alt="image" src="https://github.com/user-attachments/assets/42990713-04da-479f-96b1-954423913ad3">